File: app/src/main/java/org/zwsmith/myapplication/core/interactors/CreateUserInteractor.java
package org.zwsmith.myapplication.core.interactors;

import org.zwsmith.myapplication.core.repositories.UserRepository;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class CreateUserInteractor {
    private final UserRepository userRepository;

    @Inject
    public CreateUserInteractor(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public void createUser(String userId, String displayName, String email) {
        userRepository.createUser(userId, displayName, email);
    }
}

File: app/src/main/java/org/zwsmith/myapplication/core/interactors/CreateUserInteractor.kt
package org.zwsmith.myapplication.core.interactors

import com.google.firebase.auth.FirebaseUser
import org.zwsmith.myapplication.core.repositories.UserRepository

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class CreateUserInteractor @Inject
constructor(private val userRepository: UserRepository) {

    fun createUser(firebaseUser: FirebaseUser) {
        userRepository.createUser(firebaseUser)
    }
}
